tools/codegen/gen_backend_stubs.py

```diff
@@ -0,0 +1,114 @@
+import pathlib
+import argparse
+import os
+import yaml
```

Comment:
For ease of in tree development, we're probably going to want to setup some tests that exercise this code path on simplified YAML files that cover important functionality.

Reply:
In particular, unlike in tree native_functions.yaml, we are going to need testing for error cases and making sure we give appropriate errors in those cases. Expect tests can help make this painless.